component/src/main/java/com/celements/structEditor/fields/HiddenTagPageType.java

```diff
@@ -42,9 +42,15 @@ public void collectAttributes(AttributeBuilder attrBuilder, DocumentReference ce
     attrBuilder.addNonEmptyAttribute("type", "hidden");
     try {
       XWikiDocument cellDoc = modelAccess.getDocument(cellDocRef);
-      attrBuilder.addNonEmptyAttribute("name", modelAccess.getFieldValue(cellDoc, FIELD_NAME).or(
-          cellDocRef.getName()));
-      attrBuilder.addNonEmptyAttribute("value", getVelocityFieldValue(cellDoc, FIELD_VALUE).or(""));
+      String value = getVelocityFieldValue(cellDoc, FIELD_VALUE).or("");
+      String name = modelAccess.getFieldValue(cellDoc, FIELD_NAME).or(cellDocRef.getName());
```

Comment:
only declare `name` and `value` here, move the initialization to the else block.
`getVelocityFieldValue` might be expensive because it uses the VelocityEngine to parse and evaluate the string. it should not be executed always (even if the if condition is true).